src/main/java/com/google/devtools/build/lib/actions/FileContentsProxy.java

```diff
@@ -21,38 +21,60 @@
 
 /**
  * In case we can't get a fast digest from the filesystem, we store this metadata as a proxy to the
- * file contents. Currently it is a pair of a relevant timestamp and a "node id". On Linux the
- * former is the ctime and the latter is the inode number. We might want to add the device number in
+ * file contents. Currently it is two timestamps and a "node id". On Linux we
+ * use both ctime and mtime and inode number. We might want to add the device number in
```

Comment:
Shouldn't mtime be included in `equals()/hashCode()`, then? And possibly `addToFingerprint()`?